src/features/learning/components/PracticeCards/PracticeMatchWordAndGIF/PracticeMatchWordAndGIF.tsx

```diff
@@ -0,0 +1,118 @@
+import {typedMemo} from "../../../../../core/utils/typedMemo";
+import React, {FC, useEffect, useState} from "react";
+import styles from "./PracticeMatchWordAndGIF.module.css";
+import clsx from "clsx";
+import {ComponentProps} from "../../../../../core/models/ComponentProps";
+import PracticeIconSVG from "../../../../../assets/images/PracticeIcon.svg"
+import {LearningBlock} from "../../LearningBlock";
+import {Word} from "../../../../../core/models/Word";
+import {Typography} from "../../../../../components/Typography";
+import {SelectGIF} from "../../SelectEntity/SelectGIF";
+import {SelectButton} from "../../SelectEntity/SelectButton";
+import {SelectState} from "../../../../../core/models/SelectState";
+
+type SelectObjectState = {
+    wordObject: Readonly<Word>;
+    state: SelectState;
+}
+
+type Props = ComponentProps & Readonly<{
+    variants: Word[];
+    variantsInOtherOrder: Word[];
+}>
+
+/** Практика "Подбери пару к словам". */
+export const PracticeMatchWordAndGIF: FC<Props> = typedMemo(function PracticeMatchWordAndGIF(props) {
+    const initialWordState: SelectObjectState[] = [...props.variantsInOtherOrder.map<SelectObjectState>(variant => {
+        return {wordObject: variant, state: "default"}
+    })]
+
+    const initialGIFState: SelectObjectState[] = [...props.variants.map<SelectObjectState>(variant => {
+        return {wordObject: variant, state: "default"}
+    })]
+
+    const [wordsState, setWordsState] = useState<SelectObjectState[]>([...initialWordState])
+    const [gifsState, setGIFsState] = useState<SelectObjectState[]>([...initialGIFState])
+    const [isBlocked, setIsBlocked] = useState<boolean>(false)
+
+
+    // TODO Скорее всего изменю алгоритм для проверки данного задания
+    const handleClickOnSelectObject = (clickWordObject: Word, selectObjectState: SelectObjectState[], setSelectObjectState: React.Dispatch<React.SetStateAction<SelectObjectState[]>>) => {
+        const toDefaultState: () => { wordObject: Readonly<Word>; state: SelectState }[] = () =>
+            selectObjectState.map(objectInState => ({wordObject: objectInState.wordObject, state: objectInState.state === "success" ? "success" : "default"}))
+
+        if (!isBlocked) {
```

Comment:
```suggestion
        if (isBlocked) {
           return
        }
        if(clickWordObject) {
           ...
        } else {
           ...
        }
```

Как уменьшить вложенность условий